### PR TITLE
🐛 fix: Nil pointer dereference with Must Bind binding

### PR DESCRIPTION
**Pull request by ItsMeSamey**
if err is nil err.Error() panics
(eg. c.Bind().Must().JSON(...) successfully binds but panics

### CODE REVIEW

**Comment by gaby** [comment]
@ItsMeSamey Anyway we could trigger this in a new unit-test?

**Comment by ItsMeSamey** [comment]
@gaby I've added a test for this

File: bind.go [hunk 1]
```diff
@@ -41,12 +41,12 @@ func (b *Bind) Must() *Bind {
 
 // Check Should/Must errors and return it by usage.
 func (b *Bind) returnErr(err error) error {
-	if !b.should {
-		b.ctx.Status(StatusBadRequest)
-		return NewError(StatusBadRequest, "Bad request: "+err.Error())
+	if err == nil || b.should {
```
Comment (gaby) [inline bind.go:44]: This should be && not ||

**Review by gaby — CHANGES_REQUESTED** [review]

**Comment by ItsMeSamey** [comment]
@gaby
I've been wondering why is `Must()` even needed, it only adds overhead so won't we be better off removing it altogether. Setting status should be user's job anyway
Should i open a discussion / issue about this?

**Comment by gaby** [comment]
> @gaby I've been wondering why is `Must()` even needed, it only adds overhead so won't we be better off removing it altogether. Setting status should be user's job anyway Should i open a discussion / issue about this?

That's a question for @efectn

**Comment by ReneWerner87** [comment]
we will discuss this again internally and then let you know which way we want to go, thank you for your work and help

**Comment by ReneWerner87** [comment]
here the last statement
https://github.com/gofiber/fiber/pull/3178#issuecomment-2435208194

we want to rename the feature so that there is no confusion with the golang concept of MUST for panics

because the methods always return an error
the feature only ensures that the response is filled with it

**Comment by ReneWerner87** [comment]
give us a little time, we are currently discussing what this could look like
whether a config or the method should be renamed and where this can be found

**Comment by ReneWerner87** [comment]
@ItsMeSamey sorry for the wait

we have decided to change the `Must` method to `WithAutoHandling

by default no errors should be added to the response, only if someone calls this additional method

can you integrate this into your PR ? and extend/customize the markdown


btw benchmark error is a false positive because of the same named benchmarks
https://github.com/benchmark-action/github-action-benchmark/issues/264

File: bind.go [hunk 2]
```diff
@@ -31,22 +31,22 @@ func (b *Bind) Should() *Bind {
 	return b
 }
 
-// Must If you want to handle binder errors automatically, you can use Must.
+// If you want to handle binder errors automatically, you can use WithAutoHandling.
```
Comment (efectn) [inline bind.go:34]: Maybe we can rename Should to WithoutAutoHandling()

Comment (efectn) [reply bind.go:34]: What do you think @gaby @ReneWerner87

File: bind.go [hunk 3]
```diff
@@ -19,34 +19,35 @@ type StructValidator interface {
 
 // Bind struct
 type Bind struct {
-	ctx    Ctx
-	should bool
+	ctx        Ctx
+	dontHandle bool
```
Comment (efectn) [inline bind.go:23]: dontHandleErrs would be better

**Review by efectn — APPROVED** [review]

**Comment by ReneWerner87** [comment]
Thx

**Review by ReneWerner87 — APPROVED** [review]

**Comment by ReneWerner87** [comment]
@ItsMeSamey pls fix the lint errors 

image

**Comment by ItsMeSamey** [comment]
@gaby
I believe this is ready to be merged now

**Review by gaby — APPROVED** [review]
👍 LGTM

**Comment by gaby** [comment]
@ReneWerner87 @efectn Please check my changes to the README.md, if all good then we can merge it.

**Pull request merged**